### DISCUSSION TITLE
CC2652 Z-Stack coordinator 3.x.0 firmware does not currently support ZGP (Zigbee Green Power) GreenPower Proxy?

### DISCUSSION THREAD

**Discussion by Hedda** [Koenkk/zigbee2mqtt#13869] [General]
Does anyone who owns ZGP (Zigbee Green Power) devices and  CC2652 Zigbee Coordinator confirm this issue with Zigbee2MQTT/Z2M?

https://github.com/Koenkk/Z-Stack-firmware/issues/318

FYI, this issue was re-opened as an issue for tracking https://github.com/Koenkk/Z-Stack-firmware when raised again by s-hadinger (who is the Tasmota developer of Zigbee2Tasmota (Z2T) project which also use this firmware -> https://tasmota.github.io/docs/Zigbee/).

> **_It appears that the GP Proxy code is not enabled in coordinator ZNP code, even in the most recent versions. It is only in the sink examples, but it won't help us here._**

> **_I tried several options : enable GP_DATA_IND callback (no effect), enable GP proxy and sink (no effect). I'm giving up_**

**Comment by Hedda** [comment]
The requirements to run a Zigbee Green Power are found in the documents from Texas Instruments:

- https://training.ti.com/zigbee-green-power-enable-battery-less-devices

- http://software-dl.ti.com/simplelink/esd/simplelink_cc26x2_sdk/2.30.00.34/exports/docs/zstack/html/zigbee/gpd_application_overview.html

- https://www.ti.com/lit/an/swra615a/swra615a.pdf

![image](https://user-images.githubusercontent.com/6320001/182648825-a122224f-e865-401d-8b6f-900778fa35ad.png)

Summery; ZGP (Zigbee Green Power), a.k.a. "Zigbee GreenPower",  is a subset feature of Zigbee that allows for extreme low-power and energy-harvesting technology to be used directly with the Zigbee stack. Using Zigbee Green Power, certain home automation wireless devices (end devices) may be powered by either different self-powered energy harvesting solutions or switches and sensors that can operate for up to a decade on a single battery coin-cell battery. Energy harvesting technology enables the maintenance-free battery-less devices (end devices) such as switches and remotes operating in a Zigbee network.

https://csa-iot.org/wp-content/uploads/2022/01/docs-14-0563-18-batt-Green-Power-Basic-specification-v1.1.1.pdf

See [zigbee-herdsman PR #92](https://github.com/Koenkk/zigbee-herdsman/pull/92) for proof-of-concept as ZGP was already supported in zigbee-herdsman / Zigbee2mqtt (Z2M) projects:

- https://github.com/Koenkk/zigbee-herdsman/pull/92
- https://github.com/Koenkk/zigbee2mqtt.io/blob/develop/docs/devices/GreenPower_On_Off_Switch.md

The most common examples of Zigbee Green Power devices for home automation are rocker wall switches. Certified ZGP products:

- https://zigbeealliance.org/zigbee_products/?product_type=certified_product&se=green+power
- https://zigbeealliance.org/news_and_articles/alliance-augments-its-green-power-program/

Most popular are devices based on Zigbee 3.0 pre-certified PTM215ZE/PTM215Z and PTM216Z modules from EnOcean:

- https://www.zigbee2mqtt.io/devices/PTM_215ZE.html
- https://www.zigbee2mqtt.io/devices/PTM_215Z.html
- https://www.zigbee2mqtt.io/devices/PTM_216Z.html

EnOcean PTM215ZE/PTM215Z and PTM216Z modules use energy converter for motion energy harvesting for battery-less operation that is sold to OEM/ODM radio modules which create products:

  - [EnOcean self-powered Zigbee product series and modules for OEM](https://www.enocean.com/en/products/enocean_modules_24ghz/)
    - [EnOcean Easyfit Single / Double Rocker Wall Switch for Zigbee - EWSSZ / EWSDZ](https://www.enocean.com/en/products/enocean_modules_24ghz/easyfit-single-double-rocker-wall-switch-for-zigbee-ewssz-ewsdz/)
    - [EnOcean Easyfit Single / Double Rocker Wall Switch for Zigbee - EWSSZG / EWSDZG](https://www.enocean.com/en/products/enocean_modules_24ghz/easyfit-single-double-rocker-wall-switch-for-zigbee-ewsszg-ewsdzg/)
    - [EnOcean PTM 215ZE / PTM 215Z (PTM215ZE/PTM215Z) module](https://www.enocean.com/en/products/enocean_modules_24ghz/ptm-215ze/)
    - [EnOcean PTM 216Z (PTM216Z) module](https://www.enocean.com/en/products/enocean_modules_24ghz/ptm-216z/)
    - [EnOcean PTM 536Z (PTM536Z) Transmitter module with enhanced security features](https://www.enocean.com/en/products/enocean_modules_24ghz/ptm-536z/)
    - [EnOcean PTM 535ZB (PTM535ZB) Transmitter module with enhanced security features with both ZGP and BLE radio](https://www.enocean.com/en/products/enocean_modules_24ghz/ptm-535z/)
    - [EnOcean PTM 535Z (PTM 535Z) Transmitter module with enhanced security features](https://www.enocean.com/en/products/enocean_modules_24ghz/ptm-535z/)
        - [EnOceanECO 200 (ECO200) Energy converter for motion energy harvesting](https://www.enocean.com/en/products/enocean_modules_24ghz/eco-200/)
    - [EnOcean 2 click Clickboard from MikroElektronika for TCM 515Z (TCM515Z)](https://www.enocean.com/en/products/enocean_modules_24ghz/enocean-2-click/)
      - [EnOcean TCM 515Z (TCM515Z) Transceiver module for programmable system components](https://www.enocean.com/en/products/enocean_modules_24ghz/tcm-515z/)

Friends of Hue partnership switches from ABB (Sweden), Illumra (USA), Kichler (USA), MAKRIS by Imoon (Italy), Vimar (Italy), Busch-Jaeger (Germany), Senic (Germany), Niko (Belgium), Sunricher (Hong Kong), Koizumi (Japan), Legrand (France), and John Lewis (UK) which I believe all use either the older EnOcean PTM 215ZE module or the newer EnOcean PTM 216Z module.

- https://zigbeealliance.org/zigbee_products/ptm215z-foh/
  - https://www.signify.com/global/our-company/news/press-releases/2018/20180830-new-switch-and-new-luminaire-partners-offer-more-choice-in-design-and-control-of-your-philips-hue-smart-lights
  - https://community.smartthings.com/t/theyre-here-new-green-energy-battery-free-wall-switches-for-hue-bridge-illumra-busch-jaeger-etc/135280
  - https://www.docs.lighting.philips.com/en_gb/oem/download/xitanium/SCP_PHIL_170248_PMT215Z_table_A03.pdf
  - https://www.docs.lighting.philips.com/en_gb/oem/download/connected-lighting-solutions/switch-commissioning-weblink1.pdf

Phoscon (dresden electronic) posted these instructions for paring Friends of Hue switches with their deCONZ solution here:

  - https://phoscon.de/en/support#pairing-friends-of-hue-switch

Examples Philips Hue compatible ZGP switches:

  - [Philips Hue Click](https://runlesswire.com/click_philips_hue/) - Same as Philips Hue Tap wall-switch below?
  - [Philips Hue Tap (Wall Switch version)](https://www.cnet.com/news/this-boring-switch-might-be-the-best-philips-hue-gadget-yet/#ftag%3DCAD590a51e)
  - [Philips Hue Tap Switch (Remote Control version) Model 9290011152X](https://zigbee.blakadder.com/Philips_9290011152X.html)
  - [Philips UID8450 single-rocker switch (UID8450/10) and UID8460 double-rocker switch (UID8460/10)](https://www.lighting.philips.com/api/assets/v1/file/content/4bbf1e4388e44f3aa8e0a9e00123308d/UID8450-UID8460_SpSh_20190125_Fin.pdf) ZigBee Green Power switch and scene selector 
  - [ABB smart light switch for “Friends of Hue”](https://new.abb.com/low-voltage/products/residential-product/friends-of-hue-smart-switch)
  - [Illumra ZBT-S1AWH (single rocker, white) for “Friends of Hue”](http://illumra.com/products/wireless-light-switches/zigbee-switch)
  - [Niko Friend of Hue Smart Switch](https://www.niko.eu/en/our-products/switches-and-sockets/friends-of-hue)
  - [Senic GIRA Friends of Hue Outdoor Switch](https://www.senic.com/friends-of-hue-outdoor-switch)
    - https://zigbee.blakadder.com/Senic_100119.html
  - Sunricher Kinetic Switch 4 (SR-ZGP2801K4-FOH)
  - Sunricher Battery Switch (SR-ZGP2801K4-FOH-E)
  - [LED TRADING Dimmer Switch](https://www.amazon.de/-/en/Philips-Friends-flush-mounted-surface-mounted-version/dp/B08GKP7HPJ)
  - [LED TRADING Zigbee Push Button Coupler](https://www.amazon.de/-/en/Philips-Friends-flush-mounted-surface-mounted-version/dp/B08K3C44J4)
  - Legrand Batteryless Switches
    - https://zigbeealliance.org/zigbee_products/batteryless-2-on-off-switch-green-power/
    - https://zigbeealliance.org/zigbee_products/batteryless-2-toggles-switch-green-power/
    - https://zigbeealliance.org/zigbee_products/batteryless-toggle-switch-green-power/
    - https://zigbeealliance.org/zigbee_products/batteryless-shutter-switch-green-power-2/
    - https://zigbeealliance.org/zigbee_products/batteryless-on-off-switch-green-power/
 
Technical and developer documentation:

- https://zigbeealliance.org/wp-content/uploads/2019/11/Green-Power-White-Paper.pdf
- https://training.ti.com/zigbee-green-power-enable-battery-less-devices
- http://software-dl.ti.com/simplelink/esd/simplelink_cc26x2_sdk/2.30.00.34/exports/docs/zstack/html/zigbee/gpd_application_overview.html
- https://www.ti.com/lit/an/swra615a/swra615a.pdf
- https://www.silabs.com/community/wireless/zigbee-and-thread/knowledge-base.entry.html/2020/02/13/zigbee_green_power-fKk4
  - https://www.silabs.com/documents/public/user-guides/ug103-15-green-power-fundamentals.pdf
  - https://www.silabs.com/documents/public/user-guides/ug392-using-sl-green-power-with-ezp.pdf
- https://www.nxp.com/docs/en/user-guide/JN-UG-3095.pdf
- https://www.qorvo.com/resources/d/qorvo-zigbee-green-power-multi-sensor-comparison-to-battery-powered-zigbee-devices-white-paper

PS: Zigbee Green Power (ZGP) V2 is also included as a subset in the Zigbee 3.0 specification (Z3), but that is based on V1 Green Power energy-harvesting technology that was a subset of the Zigbee PRO standard.

**Comment by miksen** [comment]
This seems to have been resolved in zigbee2mqtt 1.28.1:
https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.28.1

I believe this is the relevant fix:
https://github.com/Koenkk/zigbee-herdsman/pull/603

On 1.26 I was not able to pair a few Sunricher green power push buttons/switches with my Sonoff dongle E coordinator, but after upgrading they pair just fine.

**Reply by Bjk8kds** [reply]
> This seems to have been resolved in zigbee2mqtt 1.28.1: https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.28.1
> 
> I believe this is the relevant fix: [Koenkk/zigbee-herdsman#603](https://github.com/Koenkk/zigbee-herdsman/pull/603)
> 
> On 1.26 I was not able to pair a few Sunricher green power push buttons/switches with my Sonoff dongle E coordinator, but after upgrading they pair just fine.

Nope, that is just GP support for EZSP adapter, still not for CC2652 mentioned above
